src/mount/direntry_cache.h

```diff
@@ -514,25 +512,31 @@ class DirEntryCache {
 		// lookup_set_ should contain all the elements inside index_set
 		auto it = lookup_set_.lower_bound(
 		    std::make_tuple(parent_inode, ctx.uid, ctx.gid, ""), LookupCompare());
-		while (it != lookup_set_.end() &&
-		       std::make_tuple(parent_inode, ctx.uid, ctx.gid) ==
-		               std::make_tuple(it->parent_inode, it->uid, it->gid)) {
-			DirEntry *entry = std::addressof(*it);
+		while (it != lookup_set_.end() && parent_inode == it->parent_inode) {
+			if (std::make_tuple(ctx.uid, ctx.gid) ==
+			    std::make_tuple(it->uid, it->gid)) {
+				DirEntry *entry = std::addressof(*it);
+				++it;
+				erase(entry);
+				continue;
```

Comment:
Even if it's true what you say, then the if will not be self-contained in terms of logic and will depend on logic outside its block. I don't like this and therefore I prefer the current version.

Reply:
However, you're right that the block can be refactored and let only the lines:
```cpp
DirEntry *entry = std::addressof(*iter);
erase(entry);
```

And the rest of the logic outside the if block. I agree with you in that part :+1:

Reply:
Well, I tried the suggested change taking all the current implementation, and compared to it without the suggested change. It seems that erasing the ```entry```, which points to the data currently pointed by ```iter```, before moving the ```iter``` pointer causes a client crash. I'm telling this because it was the result when I tried the copy of 10000 files to the mount point with and without this change. Without the suggested change the copy went fine. Note that the create syscall triggers this ```lockAndInvalidateParent``` which is under review here.